Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Slate/FollowMeToggle.cs

```diff
@@ -1,40 +1,183 @@
 
+using Microsoft.MixedReality.Toolkit.Utilities;
 using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Collections;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.Toolkit.UI
 {
     [RequireComponent(typeof(RadialView))]
     public class FollowMeToggle : MonoBehaviour
     {
+        /// <summary>
+        /// An optional object for visualizing the 'Follow Me' mode state.
+        /// </summary>
+        public GameObject VisualizationObject
+        {
+            get { return visualizationObject; }
+            set { visualizationObject = value; }
+        }
+
         [SerializeField]
-        [Tooltip("An opional object for visualizing the 'Follow Me' mode state")]
+        [Tooltip("An optional object for visualizing the 'Follow Me' mode state.")]
         private GameObject visualizationObject = null;
 
+        /// <summary>
+        /// An optional Interactable to select/deselect when toggling the follow behavior.
+        /// </summary>
+        public Interactable InteractableObject
+        {
+            get { return interactableObject; }
+            set { interactableObject = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("An optional Interactable to select/deselect when toggling the follow behavior.")]
+        private Interactable interactableObject = null;
+
+        /// <summary>
+        /// Should following be automatically enabled when the user is further than a certain distance away?
+        /// </summary>
+        public bool AutoFollowAtDistance
+        {
+            get { return autoFollowAtDistance; }
+            set
+            {
+                if (autoFollowAtDistance != value)
+                {
+                    autoFollowAtDistance = value;
+
+                    if (autoFollowAtDistance)
+                    {
+                        autoFollowDistanceCheck = StartCoroutine(AutoFollowDistanceCheck());
+                    }
+                    else
+                    {
+                        StopCoroutine(autoFollowDistanceCheck);
+                    }
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("Should following be automatically enabled when the user is further than a certain distance away?")]
+        private bool autoFollowAtDistance = false;
+
+        /// <summary>
+        /// If autoFollowAtDistance is enabled, what distance to trigger auto following at.
+        /// </summary>
+        public float AutoFollowDistance
+        {
+            get { return autoFollowDistance; }
+            set { autoFollowDistance = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("If autoFollowAtDistance is enabled, what distance to trigger auto following at.")]
+        private float autoFollowDistance = 2.0f;
+
+        /// <summary>
+        /// Optional transform to use when using autoFollowAtDistance. If not specified the local transform is used.
```

Comment:
That name is a bit confusing to me. I *think* it means "transform to modify when doing the follow behavior". Maybe AutoFollowTransformTarget?

Reply:
AutoFollowTransformTarget sounds much better! Updated.